### ISSUE TITLE
fix(common): guarantee argmax lowest-index tie-break for class labels

### ISSUE THREAD

**Issue by diegoveme** [ZKML-Soroban/ZKML-Soroban#64]
## Summary

`argmax` is the function that turns a multi-output MLP layer into a class label, and it will be the basis of the class-label public input in #43. On tied logits its result is currently unspecified: the implementation returns the last maximal index, while the common convention (and what a verifier or a reference framework will expect) is the lowest index. Because this label is meant to be reproduced on-chain and compared against a committed value, the tie-break has to be pinned down, not left to an implementation detail.

## Problem

In `crates/zkml-common/src/inference.rs`:

```rust
pub fn argmax(values: &[FixedPoint]) -> Option<usize> {
    values.iter().enumerate().max_by_key(|(_, v)| v.value).map(|(i, _)| i)
}
```

`Iterator::max_by_key` returns the last element among equal keys, so a tie resolves to the highest index. Most argmax conventions (numpy, common ML frameworks) return the first (lowest) index. The existing property test only checks that the result is stable and is a maximum, not which index wins on a tie, so the behavior is untested and unspecified.

## Scope

- Decide and document the tie-break rule (recommended: lowest index wins, to match numpy and typical decision layers).
- Adjust `argmax` so ties resolve to the lowest index deterministically.
- Add unit tests for the all-equal case and for a tie between two non-adjacent indices.
- Add a rustdoc line stating the tie-break guarantee, since downstream code (the decision layer in #43) will depend on it.

## Design notes

- Lowest-index-wins is the safer default: it matches the tools a model was likely trained and validated with, so the on-chain label agrees with the off-chain reference.
- Whatever rule is chosen must be documented alongside the public-input layout so the prover and verifier cannot disagree.

## Acceptance criteria

- [ ] `argmax` resolves ties to a single, documented index.
- [ ] Tests cover all-equal and non-adjacent-tie inputs.
- [ ] The tie-break guarantee is stated in the `argmax` rustdoc.

Related: this is a prerequisite for the class-label public input in #43.

**Comment by Anichris-koded** [comment]
Hi, my name is Koded, and I’m a professional software engineer experienced in algorithmic correctness, data processing, and reliable backend systems. I can ensure `argmax` consistently applies a lowest-index tie-break when multiple class labels share the same maximum value. My approach focuses on deterministic behavior, clear implementation, edge-case coverage, and comprehensive testing to guarantee predictable classification results.

**Comment by Peculiarr7** [comment]
Hi! I came across this issue and would love to work on it. I have experience with similar tasks and I'm confident I can investigate, implement a clean solution, and thoroughly test it before submitting a PR. If the issue is still open, I would appreciate it if you could assign it to me. I’ll prioritize this immediately and ensure a reliable solution is delivered as soon as possible. Thank you!

**Comment by akinboyewaSamson** [comment]
Hello maintainer, I would love to work on this
